src/factories/nodes.ts

```diff
@@ -213,33 +229,44 @@ export async function nodesContainerFactory(runId: string) {
     setPositions()
   }
 
-  function getActualPosition(position: Pixels): Pixels {
+  function getActualPosition(position: NodeLayoutResponse): Pixels {
     const y = rows.getTotalOffset(position.y)
-    const { x } = position
+    const x = getActualXPosition(position)
 
     return {
       x,
       y,
     }
   }
 
+  function getActualXPosition(position: NodeLayoutResponse): number {
+    if (layout.horizontal === 'dependency') {
+      return position.x + position.column * config.styles.columnGap
+    }
+
+    return position.x
+  }
+
   function resized(): void {
     const height = getHeight()
 
     container.emit('resized', { height })
   }
 
   function getHeight(): number {
-    // todo: this should probably come from the layout itself
-    let maxRow = 0
-
-    for (const [, position] of layout) {
-      maxRow = Math.max(maxRow, position.y)
+    if (!nodesLayout) {
+      return 0
     }
 
-    const height = rows.getTotalValue(maxRow)
+    return rows.getTotalValue(nodesLayout.maxRow)
+  }
+
+  function getWidth(): number {
+    if (!nodesLayout) {
+      return 0
+    }
 
-    return height
+    throw new Error('Not implemented')
```

Comment:
Placeholder. I move maxRow into the layout worker and realized that maxColumn would be helpful for calculating the width of a run. But I'll finish that in a future PR.